flexget/utils/archive.py

```diff
@@ -0,0 +1,186 @@
+        return self.archive.infolist()
+
+    def open(self, member):
+        """Returns file-like object from where the data of a member file can be read."""
+        return self.archive.open(member)
+
+    def extract_file(self, member, destination):
+        """Extract a member file to the specified destination"""
+        with self.open(member) as source:
+            with open(destination, 'wb') as target:
+                shutil.copyfileobj(source, target)
+
+        log.verbose('Extracted: %s', member)
+
+class RarArchive(Archive):
+    """
+    Wrapper class for rarfile.RarFile
+    """
+
+    def __init__(self, path):
+        if not rarfile:
+            raise NeedRarFile('Python module rarfile needed to handle RAR archives')
+
+        try:
+            super(RarArchive, self).__init__(rarfile.RarFile, path)
+        except rarfile.BadRarFile as error:
+            raise BadArchive(error)
+        except rarfile.NeedFirstVolume as error:
+            raise NeedFirstVolume(error)
+        except rarfile.Error as error:
+            raise ArchiveError(error)
+
+    def volumes(self):
+        """Returns the list of volumes that comprise this archive"""
+        return self.archive.volumelist()
+
+    def open(self, member):
+        """Returns file-like object from where the data of a member file can be read."""
+        try:
+            return super(RarArchive, self).open(member)
+        except rarfile.Error as error:
+            raise ArchiveError(error)
+
+class ZipArchive(Archive):
+    """
+    Wrapper class for zipfile.ZipFile
+    """
+
+    def __init__(self, path):
+        try:
+            super(ZipArchive, self).__init__(zipfile.ZipFile, path)
+        except zipfile.BadZipfile as error:
+            raise BadArchive(error)
+
+    def open(self, member):
+        """Returns file-like object from where the data of a member file can be read."""
+        try:
+            return super(ZipArchive, self).open(member)
+        except zipfile.BadZipfile as error:
+            raise ArchiveError(error)
+
+
+def open_archive(archive_path):
+    """
+    Returns the appropriate archive object
+    """
+
+    archive = None
+
+    if zipfile.is_zipfile(archive_path):
+        archive = ZipArchive(archive_path)
+        log.debug('Successfully opened ZIP: %s', archive_path)
+    elif rarfile and rarfile.is_rarfile(archive_path):
+        archive = RarArchive(archive_path)
+        log.debug('Successfully opened RAR: %s', archive_path)
+    else:
+        if not rarfile:
+            log.warning('Rarfile module not installed; unable to handle RAR archives.')
+
+    return archive
+
+def is_archive(path):
+    """
+    Attempts to open an entry as an archive; returns True on success, False on failure.
+    """
+
+    archive = None
+    ret = False
+
+    try:
+        archive = open_archive(path)
+        if archive:
+            archive.close()
+            ret = True
```

Comment:
Might as well just use `return True`. You're using 3 lines to do what 2 lines could.